pkg/humio/client.go

```diff
@@ -109,65 +100,22 @@ func (c *Client) GraphQLQuery(query interface{}, variables map[string]interface{
 	return client.Query(context.Background(), query, variables)
 }
 
-func NewClient(config Config) (*Client, error) {
+func NewClient(config Config, httpOpts httpclient.Options) (*Client, error) {
 	client := &Client{
 		URL: config.Address,
 	}
-	c, err := newHTTPClientWithHeaders(config)
-	if err != nil {
-		return nil, err
-	}
-	client.HTTPClient = c
-	return client, nil
-}
 
-// set up an https server and connect with tls settings. if it returns a 404 tls settings worked
-func newHTTPClientWithHeaders(config Config) (*http.Client, error) {
-	headers := map[string]string{
-		"Authorization": fmt.Sprintf("Bearer %s", config.Token),
-		"Content-Type":  "application/json",
+	httpOpts.Headers["Content-Type"] = "application/json"
+	if !httpOpts.ForwardHTTPHeaders {
+		httpOpts.Headers["Authorization"] = fmt.Sprintf("Bearer %s", config.Token)
 	}
 
-	tlsConfig, err := getTLSConfig(config)
+	c, err := httpclient.NewProvider().New(httpOpts)
 	if err != nil {
 		return nil, err
 	}
-
-	rt := http.DefaultTransport.(*http.Transport).Clone()
-	rt.TLSClientConfig = tlsConfig
-
-	return &http.Client{
-		Transport: &HttpHeaderTransport{
-			rt:      rt,
-			headers: headers,
-		},
-	}, nil
-}
-
-// getTLSConfig returns tlsConfig from settings
-// logic reused from https://github.com/grafana/grafana/blob/615c153b3a2e4d80cff263e67424af6edb992211/pkg/models/datasource_cache.go#L211
-func getTLSConfig(config Config) (*tls.Config, error) {
```

Comment:
How is TLS being handled now?

Reply:
My understanding is that TLS is handled by the default HTTP client [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/30591d3456466e8e7004ef4cbb5427e0a618066f/backend/httpclient/http_client.go#L99) once a new client is created [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/30591d3456466e8e7004ef4cbb5427e0a618066f/backend/httpclient/http_client.go#L20).